enhancements/network/communication-flows-matrix-ingress.md

```diff
@@ -0,0 +1,250 @@
+```
+
+For example,
+`ingress,TCP,6443,openshift-monitoring,prometheus-adapter,prometheus-adapter-54bb854c4f-29l6v,prometheus-adapter,worker,false`
+
+Optionally, a user can set the `custom-entries-file` flag to describe the location
+of a file with custom entires to be added to the communication matrix (can be set 
+in the supported formats), for example:
+```
+[
+  {
+    "direction": "ingress",
+    "protocol": "TCP",
+    "port": 22,
+    "service": "sshd",
+    "nodeRole": "master",
+    "optional": true
+  },
+  {
+    "direction": "ingress",
+    "protocol": "TCP",
+    "port": 51035,
+    "service": "rpc.statd",
+    "nodeRole": "master",
+    "optional": true
+  }
+]
+```
+
+Periodic tests will be added to `openshift-tests` to validate an up-to-date 
+generated communication matrix matches the communication matrix documented
+in OpenShift release documents.
+
+Another test will be added to validate the ports in a generated communication 
+matrix match a snapshot of the node's listening ports (created with the Linux
+`ss` utility).
+
+A user will be able to run `openshift-tests` or a new `oc` command, 
+`oc adm communication-matrix validate`, to validate the `EndpointSlices` 
+in the cluster match a current snapshot of the node's listening ports.
+
+### Risks and Mitigations
+
+For egress flows support (out of scope) an API change might be needed, a change that can
+affect the work done for ingress flows support.
+`EndpointSlices` stand better for ingress traffic than for egress, in ingress traffic the 
+services are less dynamic and mostly stay up during the entire cluster lifetime.
+Supporting egress traffic might require changes in the API that should also be reviewed
+and agreed with the upstream Kubernetes community.
+
+### Drawbacks
+N/A
+
+## Open Questions
+
+1. Is there a better OpenShift object to use than `EndpointSlices` to collect 
+   information about ingress traffic?
+
+2. Should a new OpenShift object be introduced to act as the source of truth
+   for cluster traffic?
+
+3. The node hosted services such as sshd, rpc, etc. are currently missing 
+   `EndpointSlices`. We believe it should be created by the Machine API as 
+    part of adding/removing nodes. This can be addressed in later enhancement.
+
+## Test Plan
+
+- Unit tests coverage
+
+- E2E tests will be added to `openshift-tests`
+  - Validate an up-to-date generated communication matrix matches the 
+    one documented in OpenShift release documents
+  - Validate the ports in a generated communication matrix match a snapshot
+    of the node's listening ports (created with the Linux `ss` utility)
+  - Use the communication matrix to create and apply firewall rules, and 
+    run E2E `openshift-tests`
+
+## Graduation Criteria
+
+### Dev Preview -> Tech Preview
+N/A
+
+### Tech Preview -> GA
+GA: 4.16
+
+### Removing a deprecated feature
+N/A
+
+## Upgrade / Downgrade Strategy
+N/A
+
+## Version Skew Strategy
+N/A
```

Comment:
Will the new command work with older clusters?

Reply:
It should.

@yuvalk what do you think?

Reply:
it'll work on any version with endpointslice (which are stable since 1.21)
problem is with all the exceptions (ie listening ports that are not covered by an endpointslices)